File: 867_TransposeMatrix.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution 
{
    public:
        vector<vector<int>> transpose(vector<vector<int>>& matrix) 
        {
            int row = matrix.size();
            int col = matrix[0].size();

            for(int i = 0; i < row-1; i++)
            {
                for(int j = i+1; j < col; j++)
                {
                    swap(matrix[i][j],matrix[j][i]);
                }
            }
            return matrix;
        }
};

int main()
{
    vector<vector<int>> matrix = {{1,2,3},{4,5,6},{7,8,9}};

    Solution obj;
    obj.transpose(matrix);;


    return 0;
}

File: 867_TransposeMatrix.java
import java.util.*;

class Solution 
{
    public int[][] transpose(int[][] matrix) 
    {
        int row = matrix.length;
        int col = matrix[0].length;

        int[][] transpose = new int[col][row];

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                transpose[j][i] = matrix[i][j];
            }
        }
        return transpose;
    }
}

public class 867_TransposeMatrix 
{
    public static void main(String[] args) 
    {
        int[][] matrix = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };

        Solution sol = new Solution();
        int[][] result = sol.transpose(matrix);

        for (int i = 0; i < result.length; i++) {
            for (int j = 0; j < result[0].length; j++) {
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }
    }
}
